sw/device/lib/runtime/uart_stdout.h

```diff
@@ -0,0 +1,20 @@
+// Copyright lowRISC contributors.
```

Comment:
What is your reasoning for adding another module here, rather than extending `print.h`?

That would allow you to do a function that turns a `dif_uart_t*` into a `buffer_sink_t`, keeping the API area smaller, right?

Reply:
Resolution of a circular dependency. `lib/dif` depends on `lib/base`, so we can't have it go the other way around. I totally agree this should go in `print.h` but to do that we'd need to move `print.h` (and `log.h`) out.

It probably makes sense to break `lib/base` into DIF-useable parts and everything else, but I didn't want to have to figure that out just to delete uart.h. Deleting `uart_stdout.h` should be far easier once we decide to merge it with `print.h`.

Reply:
`lib/dif` is only allowed to depend on very specific parts of `lib/base`, and that set does not include `print.h`. 

Given we'll primarily be printing via the UART DIF, it makes sense to me to move log and print out of base, so the layering is more reasonable. This should also mean there aren't dependencies from `lib/base` to `lib/device`. Given this PR is not urgent like a DIF is, it makes sense to do these changes as part of this PR, but not in the same commit.

Reply:
We could do that. Where do you think it makes sense for print/log to land? I was thinking `lib/runtime`.

Reply:
Let's put it in `lib` today, `lib/runtime` seems much more architecture-specific than log/print (given you plan to move check.h, what's left will be hart.h and ibex.h)

Reply:
I made a separate PR, because this PR is complicated enough and cherry-picking the change in seemed like more work than it was worth: https://github.com/lowRISC/opentitan/pull/3543

Reply:
Oh, I missed this comment. I really don't want to add more files to `lib` at the top level, since all the other libraries at the top-level there are "old" libraries that are going away, and I've been trying to empty it out. I'd prefer to move them into a subdirectory, because then the layering becomes much clearer.

(Honestly, I feel like `lib/runtime`'s "architecture-specific" meaning isn't incredibly useful, and it's really just a dumping ground for stuff we're not ok with having in `lib/base`.)

Reply:
Ok, I'm happy to accept that logging is machine-specific enough for `lib/runtime`, but I don't want that directory to turn into the same dumping ground that `lib` has become, which is a risk at the moment.